tests/shared/test_back_link.py

```diff
@@ -0,0 +1,80 @@
+
+        response = flask_test_client.get(url_for("index_bp.go_back"))
+        assert response.status_code == 302
+        assert response.headers["Location"] == url_for("fund_bp.view_all_funds")
+
+
+@pytest.mark.usefixtures("set_auth_cookie", "patch_validate_token_rs256_internal_user")
+def test_session_track_visited_pages_template(flask_test_client, seed_dynamic_data):
+    with flask_test_client.session_transaction():
+        test_form = seed_dynamic_data["forms"][0]
+        # Simulate visiting a page
+        flask_test_client.get(url_for("template_bp.view_templates"))
+        flask_test_client.get(url_for("template_bp.template_details", form_id=test_form.form_id))
+        flask_test_client.get(url_for("template_bp.edit_template", form_id=test_form.form_id))
+        assert len(session["visited_pages"]) == 3
+        assert session["visited_pages"][0]["endpoint"] == "template_bp.view_templates"
+        assert session["visited_pages"][1]["endpoint"] == "template_bp.template_details"
+        assert session["visited_pages"][2]["endpoint"] == "template_bp.edit_template"
+
+        response = flask_test_client.get(url_for("index_bp.go_back"))
+        assert response.status_code == 302
+        assert response.headers["Location"] == url_for("template_bp.template_details", form_id=test_form.form_id)
+
+        response = flask_test_client.get(url_for("index_bp.go_back"))
+        assert response.status_code == 302
+        assert response.headers["Location"] == url_for("template_bp.view_templates")
+
+
+@pytest.mark.usefixtures("set_auth_cookie", "patch_validate_token_rs256_internal_user")
+def test_session_track_visited_pages_reset_session(flask_test_client, seed_dynamic_data):
+    with flask_test_client.session_transaction():
+        # Simulate visiting a page and reset once go into view_all_funds
+        _visit_template(flask_test_client, seed_dynamic_data)
+        assert len(session["visited_pages"]) == 3
+        flask_test_client.get(url_for("fund_bp.view_all_funds"))
+        assert len(session["visited_pages"]) == 1
+        assert session["visited_pages"][0]["endpoint"] == "fund_bp.view_all_funds"
+
+        # Simulate visiting a page and reset once go into view_templates
+        _visit_template(flask_test_client, seed_dynamic_data)
+        assert len(session["visited_pages"]) == 3
+        flask_test_client.get(url_for("template_bp.view_templates"))
+        assert len(session["visited_pages"]) == 1
+        assert session["visited_pages"][0]["endpoint"] == "template_bp.view_templates"
+
+        # Simulate visiting a page and reset once go into view_templates
+        _visit_template(flask_test_client, seed_dynamic_data)
+        assert len(session["visited_pages"]) == 3
+        flask_test_client.get(url_for("round_bp.view_all_rounds"))
+        assert len(session["visited_pages"]) == 1
+        assert session["visited_pages"][0]["endpoint"] == "round_bp.view_all_rounds"
```

Comment:
Please can we parameterise this rather than test three things in a single test.

I think parameters of `target_endpoint` should be fine?

```suggestion
@pytest.mark.parametrize("expected_endpoint", ["fund_bp.view_all_funds", "template_bp.view_templates", "round_bp.view_all_rounds"])
@pytest.mark.usefixtures("set_auth_cookie", "patch_validate_token_rs256_internal_user")
def test_session_track_visited_pages_reset_session(flask_test_client, seed_dynamic_data, expected_endpoint):
    with flask_test_client.session_transaction():
        # Simulate visiting a page and reset once go into view_all_funds
        _visit_template(flask_test_client, seed_dynamic_data)
        assert len(session["visited_pages"]) == 3
        flask_test_client.get(url_for(expected_endpoint))
        assert len(session["visited_pages"]) == 1
        assert session["visited_pages"][0]["endpoint"] == expected_endpoint
```

Reply:
sure I will simplify this

Reply:
Fixed